hydromt_fiat/workflows/exposure_vector.py

```diff
@@ -957,22 +958,23 @@ def setup_ground_elevation(
                 exposure_geoms=self.get_full_gdf(self.exposure_db),
             )
             # Unit conversion
-            if (unit == "meters" or unit == "m") and (
-                self.unit == "feet" or self.unit == "ft"
-            ):
-                self.exposure_db["Ground Elevation"] = self.exposure_db[
-                    "Ground Elevation"
-                ].apply(lambda x: x * 3.28084)
-
-            elif (unit == "feet" or unit == "ft") and (
-                self.unit == "meters" or self.unit == "m"
-            ):
-                self.exposure_db["Ground Elevation"] = self.exposure_db[
-                    "Ground Elevation"
-                ].apply(lambda x: x / 3.28084)
-            else:
-                self.logger.warning(
-                    "The elevation unit is not valid. Please provide the unit of your ground elevation in 'meters' or 'feet'"
+            if unit != self.unit:
+                if (unit == Units.meters.value) and (
+                    self.unit == Units.feet.value
+                ):
+                    self.exposure_db["Ground Elevation"] = self.exposure_db[
+                        "Ground Elevation"
+                    ].apply(lambda x: x * Conversion.meters_to_feet.value)
```

Comment:
I shouldve seen this earlier, but I think the conversions need to be switched around. 
self.units = current, unit = next
so this  should convert from feet to meters